tools/storm/rollback/trident.go

```diff
@@ -0,0 +1,132 @@
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	stormrollbacktests "tridenttools/storm/rollback/tests"
+	stormrollbackconfig "tridenttools/storm/rollback/utils/config"
+	stormvmazure "tridenttools/storm/utils/vm/azure"
+	stormvmconfig "tridenttools/storm/utils/vm/config"
+	stormvmqemu "tridenttools/storm/utils/vm/qemu"
+
+	"github.com/microsoft/storm"
+
+	"github.com/sirupsen/logrus"
+)
+
+type TridentRollbackScenario struct {
+	args TridentRollbackScenarioArgs
+}
+
+type TridentRollbackScenarioArgs struct {
+	stormrollbackconfig.TestConfig `embed:""`
+	stormvmconfig.VMConfig         `embed:""`
+	stormvmqemu.QemuConfig         `embed:""`
+	stormvmazure.AzureConfig       `embed:""`
+	TestCaseToRun                  string `help:"Name of the test case to run. If not specified, all test cases will be run." default:"all"`
+}
+
+func (s *TridentRollbackScenario) Name() string {
+	return "rollback"
+}
+
+func (s *TridentRollbackScenario) Args() any {
+	return &s.args
+}
+
+func (s *TridentRollbackScenario) Tags() []string {
+	return []string{}
+}
+
+func (s *TridentRollbackScenario) StagePaths() []string {
+	return []string{}
+}
+
+func (s *TridentRollbackScenario) RegisterTestCases(r storm.TestRegistrar) error {
+	r.RegisterTestCase("prepare-qcow2", s.prepareQcow2)
+	r.RegisterTestCase("deploy-vm", s.deployVm)
+	r.RegisterTestCase("check-deployment", s.checkDeployment)
+	r.RegisterTestCase("update-and-rollback", s.updateAndRollback)
+	r.RegisterTestCase("collect-logs", s.collectLogs)
+	r.RegisterTestCase("cleanup-vm", s.cleanupVm)
+	return nil
+}
+
+func (s *TridentRollbackScenario) RequiredFiles() []string {
+	return nil
+}
+
+func (s TridentRollbackScenario) Setup(ctx storm.SetupCleanupContext) error {
+	return nil
+}
+
+func (s *TridentRollbackScenario) Cleanup(ctx storm.SetupCleanupContext) error {
+	if s.args.TestConfig.ForceCleanup {
+		// Best effort to clean up azure resources in case there was a failure
+		stormrollbacktests.CleanupVM(
+			s.args.TestConfig,
+			stormvmconfig.AllVMConfig{
+				VMConfig:    s.args.VMConfig,
+				QemuConfig:  s.args.QemuConfig,
+				AzureConfig: s.args.AzureConfig,
+			})
+	}
+	return nil
+}
+
+func (s *TridentRollbackScenario) runTestCase(tc storm.TestCase, testFunc func(stormrollbackconfig.TestConfig, stormvmconfig.AllVMConfig) error) error {
+	if tc.Name() != s.args.TestCaseToRun && s.args.TestCaseToRun != "all" {
+		tc.Skip(fmt.Sprintf("Test case '%s' does not align to TestCaseToRun '%s'", tc.Name(), s.args.TestCaseToRun))
+	} else {
+		logrus.Infof("Running test case '%s'", tc.Name())
+		// create test-specific output directory
+		testCaseSpecificConfig := s.args.TestConfig
+		testCaseSpecificConfig.OutputPath = s.args.TestConfig.OutputPath
+		if testCaseSpecificConfig.OutputPath != "" {
+			testCaseSpecificConfig.OutputPath = filepath.Join(testCaseSpecificConfig.OutputPath, tc.Name())
+			if err := os.MkdirAll(testCaseSpecificConfig.OutputPath, 0755); err != nil {
+				tc.FailFromError(err)
+			}
+		}
```

Comment:
i think this patter could be replaced/superceeded by storms artifact broker? anyway, nothing for this PR, but worth considering for the future, this seems more like something that should be upstreamed to storm